doc/src/build/programming-with-objects/ch5-dynamic-fields.md

```diff
@@ -0,0 +1,166 @@
+}
+```
+
+Now, we can define an API to add a `Child` object as a dynamic field of a `Parent` object:
+
+```rust
+public entry fun add_child(parent: &mut Parent, child: Child) {
+    ofield::add(&mut parent.id, b"child", child);
+}
+```
+
+This function takes the `Child` object by value, and makes it a dynamic field of `parent` with name `b"child"` (a byte string). At the end of the `add_child` call, we have the following ownership relationship:
+
+1. Sender address (still) owns the `Parent` object.
+2. The `Parent` object owns the `Child` object, and can refer to it by the name `b"child"`.
+
+> :warning: It is an error to overwrite a field (attempt to add a field with the same Name type and value as one that is already defined), and a transaction that does this will abort.  Fields can be modified in-place by borrowing them mutably and can be overwritten safely (e.g. to change its value type) by removing the old value first (see below for details).
+
+### Accessing Dynamic Fields
+
+Dynamic fields can be accessed by reference using the following APIs:
+
+```rust
+use sui::dynamic_field as field;
+
+public fun borrow<Name: copy + drop + store, Value: store>(
+    object: &UID,
+    name: Name,
+): &Value
+
+public fun borrow_mut<Name: copy + drop + store, Value: store>(
+    object: &mut UID,
+    name: Name,
+): &mut Value
+```
+
+Where `object` is the UID of the object the field is defined on and `name` is the field's name.
+
+> :bulb: `sui::dynamic_object_field` has equivalent functions for object fields.
+
+Let's look at how to use these APIs with the `Parent` and `Child` types defined earlier:
+
+```rust
+public entry fun mutate_child(_child: &mut Child) {}
+
+public entry fun mutate_child_via_parent(parent: &mut Parent) {
+    mutate_child(ofield::borrow_mut<vector<u8>, Child>(
+        &mut parent.id,
+        b"child",
+    ));
+}
+```
+
+The first function accepts a mutable reference to the `Child` object directly, and can be called with `Child` objects that haven't been added as fields to `Parent` objects. Its body is empty since what we care about here is not how it is mutated, but whether the function can be called at all.
+
+The second functions accepts a mutable reference to the `Parent` object and accesses its dynamic field using `borrow_mut`, to pass to `mutate_child`. This can only be called on `Parent` objects that have a `b"child"` field defined. A `Child` object that has been added to a `Parent` *must* be accessed via its dynamic field, so it can only by mutated using `mutate_child_via_parent`, not `mutate_child`, even if its ID is known.
+
+> :warning: A transaction that attempts to borrow a field that does not exist will abort.
+
+> :warning: The `Value` type passed to `borrow` and `borrow_mut` must match the type of the stored field, or the transaction will abort.
+
+> :warning: A transaction that attempts to use a dynamic object field value as an input (by value or by reference), will fail to verify and therefore to run.
```

Comment:
@tnowacki -- want to confirm with you that this property is true (I'm recalling this property from reviewing the code).

Reply:
I would maybe reword it a bit. 
"Dynamic object field values _must_ be accessed through this API. A transaction that attempts to use those objects as inputs (by value or by reference) will be marked as having invalid inputs and will be rejected"

Bit of a run-on there at the end, but the point I'm trying to get is that I don't think you will even be able to generate a certificate for the transaction.